### PR TITLE
Update loopback in package.json

### PR DESCRIPTION
**Pull request by Amir-61**
Created a loopback-2.x branch; however I do not have admin access to make it protected; @superkhau could you please make it protected? Thanks!

/to: @superkhau

### CODE REVIEW

**Comment by superkhau** [comment]
We need Juggler in the tests for this connector. See lines like https://github.com/strongloop/loopback-connector-rest/blob/dab5a1ddf9d17c8648380566d30d2a38804a62af/test/rest-adapter-custom.test.js#L9.

As for refactoring, we'll need to dig in whether only LoopBack or Juggler can be kept as a dep (depends on how the tests are written etc).

**Comment by superkhau** [comment]
I also protected the branches.

**Comment by Amir-61** [comment]
@rmg Any idea why we get this error for some machines:
```
Error: Cannot find module 'loopback-datasource-juggler'
```

**Comment by Amir-61** [comment]
@slnode test please

File: package.json [hunk 1]
```diff
@@ -32,8 +32,7 @@
     "eslint": "^2.7.0",
     "eslint-config-loopback": "^1.0.0",
     "express": "^4.12.0",
-    "loopback": "2.x.x",
-    "loopback-datasource-juggler": "2.x.x",
```
Comment (rmg) [inline package.json]: You are removing `loopback-datasource-juggler` but many tests have `require('loopback-datasource-juggler')` in them.

**Comment by rmg** [comment]
The reason only some of the platforms are failing is because npm v3 hides the dependency problem because it flattens `node_modules`.

File: package.json (hunk 1 above)
Comment (Amir-61) [inline package.json]: >@loopback-team: *IMPORTANT: In LoopBack v3.0, loopback-datasource-juggler is no longer a peer dependency that has to be explicitly listed in app/project package.json. Keep this in mind while creating `loopback-2.x` branches and updating `master` to use `loopback@3`.*

Yea I did it based on @bajtos's comment.

Shouldn't I delete `loopback-datasource-juggler` here?

Comment (superkhau) [inline package.json]: You should look at the tests and determine whether they are being used or not. If they are being used, you need to determine whether they are candidates to refactor from using loopback to juggler directly.

Comment (bajtos) [inline package.json]: My comment applies to a typical LoopBack project that calls `require('loopback')` and does not touch loopback's internals like juggler. Sorry for not making it more clear!

Most of the connectors are expected to work with vanilla juggler, therefore their tests should use juggler directly too. In which case `loopback` should not be listed in their (dev) dependencies at all.

AFAICT, `loopback` is not used in this connector at all (I run `grep "require(.loopback"` to verify).

That means we can go ahead, remove `loopback` and keep `loopback-datasource-juggler` in the dev dependencies only.

**Comment by Amir-61** [comment]
Thanks @bajtos for clarification!

@superkhau,

Updated based on Miroslav's [comment](https://github.com/strongloop/loopback-connector-rest/pull/67#discussion_r78698004). We'll see the CI result.

Thanks!

**Comment by superkhau** [comment]
test please

**Comment by superkhau** [comment]
LGTM

**Pull request merged**